rpc/handlers_test.go

```diff
@@ -1492,3 +1494,77 @@ func TestEvents(t *testing.T) {
 		require.Nil(t, events)
 	})
 }
+
+func TestAddInvokeTransaction(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	t.Cleanup(mockCtrl.Finish)
+
+	mockGateway := mocks.NewMockGateway(mockCtrl)
+	log := utils.NewNopZapLogger()
+	handler := rpc.New(nil, nil, utils.MAINNET, mockGateway, log)
+
+	t.Run("required fields are missing", func(t *testing.T) {
+		_, err := handler.AddInvokeTransaction(&gateway.BroadcastedInvokeTxn{})
+		require.NotNil(t, err)
+		assert.Equal(t, jsonrpc.InvalidParams, err.Code)
+	})
+
+	t.Run("ok", func(t *testing.T) {
+		invokeTx := generateAddInvokeTx()
+
+		txHash, err := new(felt.Felt).SetRandom()
+		require.NoError(t, err)
+		addInvokeResponse := gateway.InvokeTxResponse{
+			TransactionHash: txHash,
+		}
+
+		mockGateway.EXPECT().AddInvokeTransaction(context.TODO(), invokeTx).
+			Return(&addInvokeResponse, nil)
+
+		resp, handlerErr := handler.AddInvokeTransaction(invokeTx)
+		require.Nil(t, handlerErr)
+		assert.Equal(t, resp.TransactionHash, addInvokeResponse.TransactionHash)
+	})
+
+	t.Run("Max Fee must be non-zero", func(t *testing.T) {
+		invokeTx := generateAddInvokeTx()
+		invokeTx.MaxFee.Set(&felt.Zero)
+		resp, handlerErr := handler.AddInvokeTransaction(invokeTx)
+
+		require.Nil(t, resp)
+		assert.Equal(t, handlerErr.Code, jsonrpc.InvalidParams)
+		assert.Equal(t, handlerErr.Message, "max_fee must be bigger than 0.\n0 >= 0")
+		assert.Equal(t, handlerErr.Data, nil)
+	})
+
+	t.Run("Version must be 0x1", func(t *testing.T) {
+		invokeTx := generateAddInvokeTx()
+		invokeTx.Version = "0x0"
+		resp, handlerErr := handler.AddInvokeTransaction(invokeTx)
+
+		require.Nil(t, resp)
+		assert.Equal(t, handlerErr.Code, jsonrpc.InvalidParams)
+		assert.Equal(t, handlerErr.Message, "Transaction version '0x0' not supported. Supported versions: '0x1'")
+		assert.Equal(t, handlerErr.Data, nil)
+	})
+}
+
+func generateAddInvokeTx() *gateway.BroadcastedInvokeTxn {
```

Comment:
Is it okay to create this function that returns an invoke transaction? I find it makes the tests more readable